### PR TITLE
feat: add check for missing name/link in event document

### PR DESCRIPTION
**Pull request by maze-runnar**
For : https://github.com/fossasia/open-event-frontend/pull/7466

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/events.py [hunk 1]
```diff
@@ -540,6 +540,16 @@ def before_update_object(self, event, data, view_kwargs):
         if is_date_updated or is_draft_published or is_event_restored:
             validate_date(event, data)
 
+        if data.get('is_document_enabled'):
+            d = data.get('document_links')
+            for document in d:
+                if (document.get('name') == '') or (document.get('link')==''):
```
Comment (hongquan) [inline app/api/events.py:546]: Here, I think better write like this:

```py
if not document.get('name') or not document.get('link'):
```

**Review by hongquan — CHANGES_REQUESTED** [review]

File: app/api/events.py [hunk 2]
```diff
@@ -540,6 +540,16 @@ def before_update_object(self, event, data, view_kwargs):
         if is_date_updated or is_draft_published or is_event_restored:
             validate_date(event, data)
 
+        if data.get('is_document_enabled'):
+            d = data.get('document_links')
+            for document in d:
```
Comment (hongquan) [inline app/api/events.py:545]: What if `d` at this line is `None`?

Comment (maze-runnar) [reply app/api/events.py:545]: adding check for this

File: app/api/events.py [hunk 3]
```diff
@@ -540,6 +540,17 @@ def before_update_object(self, event, data, view_kwargs):
...
             validate_date(event, data)
 
+        if data.get('is_document_enabled'):
+            d = data.get('document_links')
+            if d:
+                for document in d:
+                    if not document.get('name') or not document.get('link'):
+                       raise UnprocessableEntityError(
+                            {'pointer': '/'},
+                            "enter required fields link and name",
```
Comment (MeghalBisht) [inline app/api/events.py:550]:
```suggestion
                            "Enter required fields link and name",
```

**Review by MeghalBisht — APPROVED** [review]

File: app/api/events.py (hunk 1 above)
Comment (maze-runnar) [reply app/api/events.py]: done

**Review by ranjitkshah — APPROVED** [review]

**Pull request merged**